lib/datadog/tracing/tracer.rb

```diff
@@ -341,12 +347,13 @@ def bind_trace_events!(trace_op)
         events = trace_op.send(:events)
 
         events.span_before_start.subscribe do |event_span_op, event_trace_op|
-          event_trace_op.service ||= @default_service
-          event_span_op.service ||= @default_service
           sample_trace(event_trace_op) if event_span_op && event_span_op.parent_id == 0
         end
 
         events.span_finished.subscribe do |event_span, event_trace_op|
+          # Fallback in case the service was never set
+          event_span.service ||= @default_service
```

Comment:
I moved the fallback `event_span.service ||= @default_service` from the span creation event to span completion evnt, because semantically a `nil` span service name has the meaning of "use whatever the default behaviour is for span service naming". But before, we were not allowing this to happen because we are setting the service name explicitly too early, in the `span_before_start` event.